airbyte-webapp/src/hooks/services/Modal/ModalService.tsx

```diff
@@ -43,7 +43,7 @@ export const ModalServiceProvider: React.FC = ({ children }) => {
         <Modal
           title={modalOptions.title}
           size={modalOptions.size}
-          onClose={() => resultSubjectRef.current?.next({ type: "canceled" })}
+          onClose={modalOptions.preventCancel ? undefined : () => resultSubjectRef.current?.next({ type: "canceled" })}
         >
           <modalOptions.content
             onCancel={() => resultSubjectRef.current?.next({ type: "canceled" })}
```

Comment:
The modal could still cancel here. Is this intended?

Reply:
In general, our modals remain unchanged in that we should be able to cancel them by clicking "cancel" or pressing escape. 

Setting `onClose` as undefined here when the `preventCancel` property is true should prevent the modal from closing via cancel.  That's what I saw when testing it manually (both by clicking outside the modal and hitting "escape" on the keyboard).  Is there something I am not thinking of there?

Reply:
I think if the implementation calls `onCancel` we would still allow to cancel the dialog, and only prevent the "self closing feature" of it. Though there might be a better property name than `preventCancel` to express that maybe, but I don't have any better ideas than `preventCancel`.